beatrix/src/test/java/org/killbill/billing/beatrix/integration/usage/TestInArrearUsagePST.java

```diff
@@ -81,4 +86,115 @@ public void testInArrearUsagePST() throws Exception {
+        }
+        catch(InvoiceApiException e) {
+            assertEquals(e.getCode(), ErrorCode.INVOICE_NOTHING_TO_DO.getCode());
+        }
+
+        //Move clock to 2024-03-21T6:40 - still no invoice
+        busHandler.pushExpectedEvents(NextEvent.NULL_INVOICE);
+        clock.addMonths(1);
+        assertListenerStatus();
+    }
+
+    @Test(groups = "slow")
+    public void testInArrearUsageCreateCancelPST() throws Exception { //PST and UTC on different days
+        DateTime today = new DateTime(2024, 2, 21, 6, 00); //2024-02-16T6:00 UTC=2024-02-15T10:30 PST (different days)
+        clock.setTime(today);
+
+        DateTime referenceTime = new DateTime(2024, 2, 20, 9, 0);
+        final AccountData accountData = getAccountData(20, DateTimeZone.forID("America/Los_Angeles"), referenceTime);
+        final Account account = createAccountWithNonOsgiPaymentMethod(accountData);
+        accountChecker.checkAccount(account.getId(), accountData, callContext);
+
+        final PlanPhaseSpecifier baseSpec = new PlanPhaseSpecifier("pistol-in-arrear-monthly-notrial");
+        final PlanPhaseSpecifier addOnSpec = new PlanPhaseSpecifier("bullets-usage-in-arrear");
+
+        final List<EntitlementSpecifier> specifierList = List.of(new DefaultEntitlementSpecifier(baseSpec), new DefaultEntitlementSpecifier(addOnSpec));
+        final BaseEntitlementWithAddOnsSpecifier cartSpecifier = new DefaultBaseEntitlementWithAddOnsSpecifier(null, "key1", specifierList, null, null, false);
+        final List<BaseEntitlementWithAddOnsSpecifier> entitlementWithAddOnsSpecifierList = List.of(cartSpecifier);
+
+        //create subscription at 6:05
+        clock.setTime(today.plusMinutes(5));
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.CREATE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        final List<UUID> allEntitlements = entitlementApi.createBaseEntitlementsWithAddOns(account.getId(), entitlementWithAddOnsSpecifierList, true, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        final Entitlement baseEntitlement = entitlementApi.getEntitlementForId(allEntitlements.get(0), false, callContext);
+        final Entitlement addOnEntitlement = entitlementApi.getEntitlementForId(allEntitlements.get(1), false, callContext);
+
+        //Record usage at 6:30
+        clock.setTime(today.plusMinutes(30));
+        recordUsageData(addOnEntitlement.getId(), "t1", "bullets", clock.getUTCNow(), BigDecimal.valueOf(85L), callContext);
+
+        //cancel base at 6:40
+        clock.setTime(today.plusMinutes(40));
+        busHandler.pushExpectedEvents(NextEvent.BLOCK, NextEvent.BLOCK, NextEvent.CANCEL, NextEvent.CANCEL, NextEvent.INVOICE, NextEvent.PAYMENT, NextEvent.INVOICE_PAYMENT);
+        baseEntitlement.cancelEntitlementWithPolicy(EntitlementActionPolicy.IMMEDIATE, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //$0 invoice corresponding to usage item
```

Comment:
Nit: not a $0 invoice any more, this comment needs to be updated to `//Invoice generated for the usage item`

Reply:
Ack - fixed in [b6cc4b5](https://github.com/killbill/killbill/pull/1982/commits/b6cc4b576a816fde8bd10820341309673c256552)